modules/potree/src/lib/potree-node-source.ts

```diff
@@ -0,0 +1,119 @@
+import {load} from '@loaders.gl/core';
+import {MeshGeometry} from '@loaders.gl/schema';
+import {DataSource, DataSourceProps, LoaderOptions, resolvePath} from '@loaders.gl/loader-utils';
+import {LASLoader} from '@loaders.gl/las';
+import {PotreeMetadata} from '../types/potree-metadata';
+import {POTreeNode} from '../parsers/parse-potree-hierarchy-chunk';
+import {PotreeHierarchyChunkLoader} from '../potree-hierarchy-chunk-loader';
+import {PotreeLoader} from '../potree-loader';
+
+export type PotreeNodesSourceProps = DataSourceProps & {
+  attributions?: string[];
+  potree?: {
+    loadOptions?: LoaderOptions; // PotreeLoaderOptions;
+    // TODO - add options here
+  };
+};
+
+/**
+ * A Potree data source
+ * @note Point cloud nodes tile source
+ */
+export class PotreeNodesSource extends DataSource {
+  url: string;
+  data: string | Blob;
+  props: PotreeNodesSourceProps;
+  metadata: PotreeMetadata | null = null;
+  root: POTreeNode | null = null;
+  isReady = false;
+
+  private initPromise: Promise<void> | null = null;
+
+  get isSupported(): boolean {
+    return (
+      this.isReady &&
+      ['1.4', '1.7'].includes(this.metadata?.version ?? '') &&
+      typeof this.metadata?.pointAttributes === 'string' &&
+      ['LAS', 'LAZ'].includes(this.metadata?.pointAttributes)
+    );
+  }
+
+  get contentExtension(): string | null {
+    if (!this.isReady) {
+      return null;
+    }
+    switch (this.metadata?.pointAttributes) {
+      case 'LAS':
+        return 'las';
+      case 'LAZ':
+        return 'laz';
+      default:
+        return 'bin';
+    }
+  }
+
+  constructor(data: string | Blob, props: PotreeNodesSourceProps) {
+    super(props);
+    this.props = props;
+    this.data = data;
+    this.url = typeof data === 'string' ? resolvePath(data) : '';
+
+    this.initPromise = this.init();
+  }
+
+  async init() {
+    this.metadata = await load(`${this.url}/cloud.js`, PotreeLoader);
```

Comment:
Is it always called `cloud.js`? Is the URL just supposed to be a directory? Maybe specify in a comment here and on the prop.

To make it easier for the user we may have logic that accepts both a filename with `cloud.js` at the end and just a directory name.

Reply:
Done